src/utils/countSuggestions.js

```diff
@@ -1,7 +1,8 @@
-const countSuggestions = suggestions =>
-  Object.values(suggestions).reduce(
-    (acc, { results }) => acc + results.length,
-    0
-  )
+import { isUndefined } from 'lodash'
+
+const countSuggestions = (suggestions, maxSuggestions) =>
+  isUndefined(maxSuggestions)
+    ? Object.values(suggestions).reduce((acc, { results }) => acc + results.length, 0)
+    : maxSuggestions
```

Comment:
I believe you should go for a `Math.min()` here. 

Currently, there will be a problem if `maxSuggestions` is defined but there are fewer suggestions than the maximum. In this case, the function won't return the actual count, but the maximum, which will break keyboard navigation again.

(We really need to increase test coverage of this package!)